src/api/apps/articles/model/save.coffee

```diff
@@ -160,13 +160,11 @@ removeStopWords = (title) ->
 
 @sanitizeAndSave = (callback) => (err, article) =>
   return callback err if err
-  if article.published or article.scheduled_publish_at
-    article = setOnPublishFields article
-    indexForSearch(article, ->) if article.indexable
-    db.articles.save sanitize(article), callback
-  else
-    indexForSearch(article, ->) if article.indexable
-    db.articles.save sanitize(article), callback
+  article = setOnPublishFields article if article.published or article.scheduled_publish_at
+  indexForSearch(article, ->) if article.indexable
+  inputData = _.omit(article, '_id')
+  db.collection('articles').updateOne { _id: article._id }, { $set: sanitize(inputData)}, { upsert: true }, (err, res) ->
+    db.collection('articles').findOne { _id: article._id || res.upsertedId }, callback
```

Comment:
From the tests my understanding is this function will need to support document creation and updating, hence the `upsert: true` option. If the passed in `article` has an `_id`, it will update the existing document, otherwise it will create a new one. 

I want to point out that the test documents have both an `_id` and `id` and they are often different (I confirmed this is true in the current main branch, also). I assume that this is a result of our test setup. To not expand the scope of this PR, I haven't addressed this and always assume the `_id` is the correct primary key.